### ISSUE TITLE
microk8s dashboard (default install) deploys invalid certificates

### ISSUE THREAD

**Issue by planetf1** [canonical/microk8s#2507]
- macOS Big Sur 11.5.1
- microk8s 2.1.0 installed from home-brew today
- 'microk8s inspect' not working (will raise a different issue on this)
- Additional features installed including 'dashboard' (no apps/customization yet)
- Run 'microk8s dashboard-proxy
```
jonesn:LaunchAgents/ $ microk8s dashboard-proxy                                                                                                                [17:20:31]
Checking if Dashboard is running.
Waiting for Dashboard to come up.
Dashboard will be available at https://192.168.64.2:10443
Use the following token to login:
eyJhbGciOiJSUzI1NiIsImtpZCI6ImZqbVM2Q0hJRXhFeDRaQmk4MGp3eXdVX01RNTZGaGhyblVSTll6NHNJWkEifQ.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJrdWJlLXN5c3RlbSIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VjcmV0Lm5hbWUiOiJkZWZhdWx0LXRva2VuLTlud2dsIiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZXJ2aWNlLWFjY291bnQubmFtZSI6ImRlZmF1bHQiLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlcnZpY2UtYWNjb3VudC51aWQiOiI1Mzc0MmY1My01NDEzLTQ5MGMtOTc2Mi0xMTIyNzZjMzQ2YzAiLCJzdWIiOiJzeXN0ZW06c2VydmljZWFjY291bnQ6a3ViZS1zeXN0ZW06ZGVmYXVsdCJ9.kNrTQJNjVK6c8zu-AWggpStD5LgTUhEcA1EHotaJEUvPWWJ-2Q6qMI1AKm1cww9Ie1jynKIprE4NSoh1ot8l9bxZbzimACFdTV6ggHcc7C4oBdCj1aYUGNQpd1DAkNXmC8IAEMEyDG0D3IBGAtxTCt_ptC86qE-l87Lu-tM5J9WgRe4jkXOfKQxVF_ctWNs9n6fh-LaYTVIe-4kM7238Fcf-cZY6THKgjg73XQOeHNGbIOA23pB3JHaD1-Y_j2xbtbpK7p32ustRsZNR3vqEpw5ldWK_Z0KTD3lbf8vV8m6vRlsBAIb9EPm4Yrkc_FWDMQkdrAbYuZdZsNAYfWvzlg
Forwarding from 0.0.0.0:10443 -> 8443
Handling connection for 10443
```

However after doing this, the page will not load in Chrome:
<img width="760" alt="Screenshot 2021-08-09 at 17 32 31" src="https://user-images.githubusercontent.com/7292002/128741171-86e4b5f4-6347-4e01-9a24-4506bf03de81.png">

nor will it load in safari:
<img width="638" alt="Screenshot 2021-08-09 at 17 31 34" src="https://user-images.githubusercontent.com/7292002/128741116-dd5f29bf-6684-4f24-a67a-20b065375bbd.png">

Getting a warning of self signed projects is ok, but here the cert cannot even be bypassed. (Been there with my own project!)

This renders the dashboard not immediately usable.

The defaults should at least generate a better cert which offers bypass in at least one well known browser, or a cert should be generated as part of the install

**Comment by ktsakalozos** [comment]
Hi @planetf1 you see this error because the dashboard shows a self signed certificate. Each browser has its own way to allow such certificates see for example [1] on how this is done on chrome.

[1] https://stackoverflow.com/questions/7580508/getting-chrome-to-accept-self-signed-localhost-certificate

**Comment by planetf1** [comment]
Hi,
 I've generated some certs for a project I work on - but cannot pretend to be an expert...
 
 I noticed exactly how the cert was generated governs whether chrome or safari permit a simple-is bypass. I've not checked in detail the microk8s dashboard cert but wonder if it needs to be modified slightly to at least go down the bypass route. I think (but can't be 100%) it was subjectAltName having a value (for example localhost) allowed the bypass with current browser versions - even if wrong. Of course this is a moving target.
 
 So even if the cert were self signed, if this was corrected (hopefully subjectAltName) then the user could at least click advanced/bypass & continue to the dashboard. 
 
 Alternatively the install docs should make it clear they need to gen a cert (with guidance on how/link to article), or generate a cert for them.
 
 I think this is important since one of the key USPs of microk8s bundling on macOS etc is ease of use?

 I can take a look at the certs in more detail if it helps - I struggled with this on my own project ( https://github.com/odpi/egeria )
